doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        Installing,
+        /// The installer has completed and cleanup actions are in progress. Cancellation of the 
+        /// IAsyncOperationWithProgress in this state will not stop cleanup or roll back the install.
+        PostInstall,
+        /// The operation has completed.
+        Finished,
+    };
+
+    /// Progress object for the install
+    /// DESIGN NOTE: percentage for the install as a whole is purposefully not included as there is no way to 
+    /// estimate progress when the installer is running.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    struct InstallProgress
+    {
+        /// State of the install
+        AppInstallProgressState State;
+        /// DESIGN NOTE: BytesDownloaded may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes downloaded if known
+        UInt64 BytesDownloaded;
+        /// DESIGN NOTE: BytesRequired may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes required if known
+        UInt64 BytesRequired;
+        /// Download percentage completed
+        Single DownloadPercentage;
+        /// Install percentage if known.
+        Single InstallPercentage;
+    };
+
+    /// Result of the install
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass InstallResult
+    {
+        /// Used by a caller to correlate the install with a caller's data.
+        String CorrelationData{ get; };
+        /// Whether a restart is required to complete the install.
+        Boolean RebootRequired { get; };
+    }
+
+    /// IMPLEMENTATION NOTE: SourceOrigin from AppInstallerRepositorySource.h
+    /// Defines the origin of the app catalog details.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogOrigin
+    {
+        /// Predefined means it came as part of the Windows Package Manager package and cannot be removed.
+        Predefined,
+        /// User means it was added by the user and could be removed.
+        User,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceTrustLevel from AppInstallerRepositorySource.h
+    /// Defines the trust level of the app catalog.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppCatalogInfo
+    {
+        /// The app catalog's unique identifier.
+        String Id { get; };
+        /// The name of the app catalog.
+        String Name { get; };
+        /// The type of the app catalog.
+        String Type { get; };

```

Comment:
The two valid values right now are "Microsoft.PreIndexed.Package" and "Microsoft.Rest". I can turn that into an enum but it will mean having to rev the interface every time one is added and I'm not sure how infrequent that will be. As an enum this might become the most likely thing to cause us to need a V2.